mypy/newsemanal/semanal.py

```diff
@@ -2369,7 +2369,9 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
         # so we need to replace it with non-explicit Anys.
         res = make_any_non_explicit(res)
         no_args = isinstance(res, Instance) and not res.args
-        fix_instance_types(res, self.fail)
+        fix_instance_types(res, self.fail,
```

Comment:
I think this needs to be done conditionally only when `no_args`, maybe?

Reply:
Yes, aliases like `A = List` should be allowed even with the flag, will fix it now.